containers/src/Kokkos_DynRankView.hpp

```diff
@@ -478,9 +483,17 @@ class DynRankView : private View<DataType*******, Properties...> {
   // Types below - at least the HostMirror requires the value_type, NOT the rank
   // 7 data_type of the traits
 
+#ifdef KOKKOS_ENABLE_DEPRECATED_CODE_4
   /** \brief  Compatible view of array of scalar types */
-  using array_type = DynRankView<
-      typename drvtraits::scalar_array_type, typename drvtraits::array_layout,
+  using array_type KOKKOS_DEPRECATED_WITH_COMMENT(
+      "Use Kokkos_DynRankView::type insted.") =
+      DynRankView<
+          typename drvtraits::data_type, typename drvtraits::array_layout,
+          typename drvtraits::device_type, typename drvtraits::memory_traits>;
+#endif
+  /** \brief  Compatible view of data type */
+  using type = DynRankView<
+      typename drvtraits::data_type, typename drvtraits::array_layout,
       typename drvtraits::device_type, typename drvtraits::memory_traits>;
```

Comment:
Is this just supposed to be an alias to the current class type? Why do we need this?

Reply:
Yes. It replaces a public type so having an equivalent public replacement might be good.

Reply:
Why are you deprecating it then? Does it conflict with something?
If it's used internally, does it need to be public?

Reply:
ping